src/physics/euler.cpp

```diff
@@ -390,16 +384,18 @@ template<typename real2>
 inline real2 Euler<dim,nstate,real>
 :: compute_entropy (const real2 density, const real2 pressure) const
 {
-    real2 entropy = pressure * pow(density, -gam);
-    if (entropy > 0){
-        entropy = log( entropy );
+    // copy density and pressure such that the check will not modify originals
+    real2 density_check = density; 
+    real2 pressure_check = pressure;
+    const bool density_is_positive = check_positive_quantity(density_check, "density");
+    const bool pressure_is_positive = check_positive_quantity(pressure_check, "pressure");
+    if (density_is_positive && pressure_is_positive) {
+        real2 entropy = pressure * pow(density, -gam);
+        entropy = log(entropy);
+        return entropy;
     } else {
-        entropy = BIG_NUMBER;
-        this->pcout << "WARNING: Entropy is not defined because " << std::endl 
-                    << "    pressure * pow(density, -gam) < 0 ." << std::endl
-                    << "    Setting entropy = BIG_NUMBER." << std::endl;
+        return BIG_NUMBER;
```

Comment:
also, shouldn't the function in physics base be handling all instances where we set it to `BIG_NUMBER`?

Reply:
In principle, I agree. However, this was the best way I could think of to replicate the original behaviour of the code. I could call `handle_non_physical_result` from this else clause, but then it will print warnings twice.

Let me know if you have ideas for how to execute this better.